### PR TITLE
Refactor eigenwave script

### PR DESCRIPTION
**Pull request by mlange05**
This merge refactors the eigenwave3d.py example script to provide a ncie and clean user interface, including automatic parallel execution and an optional flag for solution output. More specifically it includes:
* Automatic parallel execution that sets `OMP_NUM_PROCS` and the affinity at run-time
* Improved user interface for the eigenwave3d.py script through use of an ArgumentParser
* New flags for test script, including a compiler flag, run-time execution and an output flag
* Updated README entries for example script and dependency installation

Note, that this wants PR #32 to land first.

### CODE REVIEW

File: tests/eigenwave3d.py [hunk 1]
```diff
@@ -224,53 +210,66 @@ def converge_test():
...
-        return
-
-    if sys.argv[1] == 'converge':
+    ModeHelp = """Avalable testing modes:
+default:   Eigenwave test case on a unit cube grid (100 x 100 x 100)
+
+read:      Test for model intialisation from input file; computes
+           eigenwave on a unit cube grid (200 x 200 x 200)
+
+converge:  Convergence test of the (2,4) scheme; halves spacing
```
Comment (ctjacobs) [inline tests/eigenwave3d.py]: What exactly do you mean by the (2,4) scheme here?

File: tests/eigenwave3d.py [hunk 2]
```diff
@@ -224,53 +210,66 @@ def converge_test():
...
+    p.add_argument('mode', choices=('default', 'read', 'converge', 'cx1'),
+                   nargs='?', default='default', help=ModeHelp)
+    p.add_argument('-c', '--compiler', default='g++',
+                   help='C++ Compiler to use for model compilation, eg. g++ or icpc')
+    p.add_argument('-x', '--execute', action='store_true', default=False,
+                   help='Dynamically execute the generated model')
+    p.add_argument('-n', '--nthreads', type=int, default=1,
+                   help='Number of threads for dynamic execution')
+    p.add_argument('-o', '--output', action='store_true', default=False,
+                   help='Activate soultion output in .vts format')
```
Comment (ctjacobs) [inline tests/eigenwave3d.py]: "soultion" -> "solution"
".vts" -> ".vtu"?

Comment (mlange05) [inline tests/eigenwave3d.py]: We do actually seem to output .vts files atm. Will fix the other typo.

**Comment by mlange05** [comment]
Help messages are updated now. Anyone opposed to merging this?

**Comment by ctjacobs** [comment]
Looks fine to me. Go.

**Pull request merged**